openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/retentionPolicyApp/RetentionPolicyApp.java

```diff
@@ -0,0 +1,121 @@
+      return;
+    }
+
+    RetentionPolicyConfiguration.Entity entity = retentionPolicyConfiguration.getEntity();
+    RetentionPolicyConfiguration.RetentionPeriod retentionPeriod =
+        retentionPolicyConfiguration.getRetentionPeriod();
+
+    switch (entity) {
+      case EVENT_SUBSCRIPTION -> eventSubscriptionCleanupHandler(retentionPeriod);
+      default -> throw new IllegalArgumentException("Unknown entity type: " + entity);
+    }
+  }
+
+  private long getRetentionPeriodInMilliseconds(
+      RetentionPolicyConfiguration.RetentionPeriod retentionPeriod) {
+    return switch (retentionPeriod) {
+      case ONE_WEEK -> getRetentionCutoffMillis(Duration.ofDays(7).toMillis());
+      case TWO_WEEKS -> getRetentionCutoffMillis(Duration.ofDays(14).toMillis());
+      case ONE_MONTH -> getRetentionCutoffMillis(Duration.ofDays(30).toMillis());
+      case THREE_MONTHS -> getRetentionCutoffMillis(Duration.ofDays(90).toMillis());
+      case SIX_MONTHS -> getRetentionCutoffMillis(Duration.ofDays(180).toMillis());
+    };
+  }
+
+  private long getRetentionCutoffMillis(long durationMillis) {
+    return Instant.now().minusMillis(durationMillis).toEpochMilli();
+  }
+
+  private void eventSubscriptionCleanupHandler(
+      RetentionPolicyConfiguration.RetentionPeriod retentionPeriod) {
+    long retentionPeriodInMilliseconds = getRetentionPeriodInMilliseconds(retentionPeriod);
+    final int BATCH_SIZE = 10_000;
+
+    // Delete old records from successful_sent_change_events
+    int totalDeletedSuccessfulEvents =
+        batchDelete(
+            () ->
+                Entity.getCollectionDAO()
+                    .eventSubscriptionDAO()
+                    .deleteSuccessfulSentChangeEventsInBatches(
+                        retentionPeriodInMilliseconds, BATCH_SIZE));
+
+    // Delete old records from change_event
+    int totalDeletedChangeEvents =
+        batchDelete(
+            () ->
+                Entity.getCollectionDAO()
+                    .eventSubscriptionDAO()
+                    .deleteChangeEventsInBatches(retentionPeriodInMilliseconds, BATCH_SIZE));
+
+    // Delete old records from consumers_dlq
+    int totalDeletedDlq =
+        batchDelete(
+            () ->
+                Entity.getCollectionDAO()
+                    .eventSubscriptionDAO()
+                    .deleteConsumersDlqInBatches(retentionPeriodInMilliseconds, BATCH_SIZE));
+
+    LOG.info(
+        "Deleted {} old successful_sent_change_events, {} old change_event, and {} old consumers_dlq records older than {} for EVENT_SUBSCRIPTION.",
+        totalDeletedSuccessfulEvents,
+        totalDeletedChangeEvents,
+        totalDeletedDlq,
+        retentionPeriod);
+  }
+
+  /**
+   * Helper method to run a chunked delete operation until no more large batches are returned.
+   * The Supplier should execute a single batch delete and return the number of records deleted.
+   */
+  private int batchDelete(Supplier<Integer> deleteFunction) {
```

Comment:
why we need this method?

Reply:
To perform large delete operations in batches rather than deleting everything at once.